civiclens/nlp/comments.py

```diff
@@ -364,9 +364,12 @@ def rep_comment_analysis(id: str, model: SentenceTransformer) -> RepComments:
         comment_data.rep_comments = form_letters + df_rep_paraphrase.to_dicts()
         comment_data.num_representative_comment = len(comment_data.rep_comments)
 
+    num_paraphrased = count_unique_comments(df_paraphrases)
     comment_data.num_total_comments = df.shape[0]
```

Comment:
Do you still have questions on this/are you working through the bug?

One thought I had when you're debugging this: line 150 in `assign_clusters` is a rows variable that equals df.shape[0]. You could return that rows variable and compare it with df.shape[0] in line 368 here to see if there is some kind of accidental transformation that is making the df change size. I'm just trying to think why df_num_paraphrase could possibly have more rows than the df it is a subset of...